### DISCUSSION TITLE
MESA GUI Editor Development

### DISCUSSION THREAD

**Discussion by ENUMERA8OR** [mesa/mesa#2159] [Ideas]
As an aspiring developer, I suggest solving the complexity and time-consuming nature of creating agent-based models in Mesa using Python code using a GUI editor. Currently, the process involves manual coding of agent-based models, which can be prone to errors and inefficiencies, especially for users who are not familiar with programming. It will simplify and automate the process of creating agent-based models in Mesa using a visual editor.

Please let me know your thoughts on this.

**Comment by rht** [comment]
That may be helpful in the pre-AI-LLM era, but nowadays you can ask language models to write ABM for you. Instead, what I find more powerful is to specify your ABM declaratively in a .yaml/.toml file, see e.g. dolo project, example model: https://github.com/EconForge/dolo.py/blob/master/examples/models/rmt3_ch11.yaml.

**Comment by wang-boyu** [comment]
Something like AnyLogic? It sounds great but may need lots of hard work.

> Instead, what I find more powerful is to specify your ABM declaratively in a .yaml/.toml file

This could be the interface between Mesa and a GUI app - essentially the GUI app is a text editor for the .yaml/.toml model definition.

**Reply by ENUMERA8OR** [reply]
Interesting, will definitely get back to you about that.
